u2f-ref-code/java/src/com/google/u2f/codec/RawMessageCodec.java

```diff
@@ -192,6 +194,85 @@ public static AuthenticateResponse decodeAuthenticateResponse(byte[] data) throw
     }
   }
 
+  public static TransferAccessResponse decodeTransferAccessResponse(byte[] data)
+      throws U2FException {
+    int sequenceNumberSize = 1;
+    int publicKeySize = 65;
+    int appIdSize = 32;
+
+    try {
+      DataInputStream inputStream = new DataInputStream(new ByteArrayInputStream(data));
+      byte controlByte = inputStream.readByte();
+
+      if (!inputStream.markSupported()) {
+        throw new U2FException("Stream doesn't support mark and reset.");
+      }
+      inputStream.mark(inputStream.available());
+      int numberOfTransferAccessMessages = inputStream.readUnsignedByte();
+      inputStream.reset();
+
+      TransferAccessMessage[] transferAccessMessages =
+          new TransferAccessMessage[numberOfTransferAccessMessages];
+
+      for (int i = 0; i < numberOfTransferAccessMessages; i++) {
```

Comment:
the body of this for loop is almost asking to be its own method.  Try it and see if it works.